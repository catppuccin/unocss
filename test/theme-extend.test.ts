import { describe, expect, test } from 'vitest';
import { createGenerator, presetMini } from 'unocss';

import { extendCatppuccin } from '../dist';

describe('Catppuccin colours with theme extender using presetMini', () => {
  test('and without default variant', async () => {
    const uno = createGenerator({
      presets: [
        presetMini({
          dark: 'media',
        }),
      ],
      theme: {
        colors: extendCatppuccin(),
      },
    });

    const { css } = await uno.generate(
      [
        'text-latte-text',
        'dark:text-mocha-text',

        'light:bg-latte-base',
        'dark:bg-mocha-base',

        'border-frappe-pink/50',
        'hover:border-frappe-pink',

        'not-disabled:bg-macchiato-red',
        'disabled:bg-macchiato-red/25',

        'not-visited:bg-mocha-blue',
        'visited:bg-mocha-flamingo',
      ],
      { preflights: false }
    );

    expect(css).toMatchSnapshot();
  });

  test('and with default macchiato variant', async () => {
    const uno = createGenerator({
      presets: [
        presetMini({
          dark: 'media',
        }),
      ],
      theme: {
        colors: extendCatppuccin({
          variant: 'macchiato',
        }),
      },
    });

    const { css: generatedCSS } = await uno.generate(
        [
          'text-text',
          'dark:text-text',

          'light:bg-base',
          'dark:bg-base',

          'border-pink/50',
          'hover:border-pink',

          'not-disabled:bg-red',
          'disabled:bg-red/25',

          'not-visited:bg-blue',
          'visited:bg-flamingo',
        ],
        { preflights: false }
      ),
      { css: blankCSS } = await uno.generate(
        [
          'text-latte-text',
          'dark:text-mocha-text',

          'light:bg-latte-base',
          'dark:bg-mocha-base',
        ],
        { preflights: false }
      );

    expect(generatedCSS).toMatchSnapshot();
    expect(blankCSS).toBe('');
  });

  describe('and with custom prefix', () => {
    test('without default variant', async () => {
      const uno = createGenerator({
        presets: [
          presetMini({
            dark: 'media',
          }),
        ],
        theme: {
          colors: extendCatppuccin({
            prefix: 'ctp',
          }),
        },
      });

      const { css: generatedCSS } = await uno.generate(
          [
            'text-ctp-latte-text',
            'dark:text-ctp-mocha-text',

            'light:bg-ctp-latte-base',
            'dark:bg-ctp-mocha-base',

            'border-ctp-frappe-pink/50',
            'hover:border-ctp-frappe-pink',

            'not-disabled:bg-ctp-macchiato-red',
            'disabled:bg-ctp-macchiato-red/25',

            'not-visited:bg-ctp-mocha-blue',
            'visited:bg-ctp-mocha-flamingo',
          ],
          { preflights: false }
        ),
        { css: blankCSS } = await uno.generate(
          [
            'text-latte-text',
            'dark:text-mocha-text',

            'light:bg-latte-base',
            'dark:bg-mocha-base',

            'text-ctp-mauve',
            'bg-ctp-surface0',
          ],
          { preflights: false }
        );

      expect(generatedCSS).toMatchSnapshot();
      expect(blankCSS).toBe('');
    });
  });
});
