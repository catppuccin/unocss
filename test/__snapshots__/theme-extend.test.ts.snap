// Vitest Snapshot v1

exports[`Catppuccin colours with theme extender using presetMini > and with custom prefix > without default variant 1`] = `
"/* layer: default */
.disabled\\\\:bg-ctp-macchiato-red\\\\/25:disabled{background-color:rgba(237,135,150,0.25);}
.not-disabled\\\\:bg-ctp-macchiato-red:not(:disabled){--un-bg-opacity:1;background-color:rgba(237,135,150,var(--un-bg-opacity));}
.not-visited\\\\:bg-ctp-mocha-blue:not(:visited){--un-bg-opacity:1;background-color:rgba(137,180,250,var(--un-bg-opacity));}
.visited\\\\:bg-ctp-mocha-flamingo:visited{--un-bg-opacity:1;background-color:rgba(242,205,205,var(--un-bg-opacity));}
.border-ctp-frappe-pink\\\\/50{border-color:rgba(244,184,228,0.5);}
.hover\\\\:border-ctp-frappe-pink:hover{--un-border-opacity:1;border-color:rgba(244,184,228,var(--un-border-opacity));}
.text-ctp-latte-text{--un-text-opacity:1;color:rgba(76,79,105,var(--un-text-opacity));}
@media (prefers-color-scheme: dark){
.dark\\\\:bg-ctp-mocha-base{--un-bg-opacity:1;background-color:rgba(30,30,46,var(--un-bg-opacity));}
.dark\\\\:text-ctp-mocha-text{--un-text-opacity:1;color:rgba(205,214,244,var(--un-text-opacity));}
}
@media (prefers-color-scheme: light){
.light\\\\:bg-ctp-latte-base{--un-bg-opacity:1;background-color:rgba(239,241,245,var(--un-bg-opacity));}
}"
`;

exports[`Catppuccin colours with theme extender using presetMini > and with default macchiato variant 1`] = `
"/* layer: default */
.disabled\\\\:bg-red\\\\/25:disabled{background-color:rgba(237,135,150,0.25);}
.not-disabled\\\\:bg-red:not(:disabled){--un-bg-opacity:1;background-color:rgba(237,135,150,var(--un-bg-opacity));}
.not-visited\\\\:bg-blue:not(:visited){--un-bg-opacity:1;background-color:rgba(138,173,244,var(--un-bg-opacity));}
.visited\\\\:bg-flamingo:visited{--un-bg-opacity:1;background-color:rgba(240,198,198,var(--un-bg-opacity));}
.border-pink\\\\/50{border-color:rgba(245,189,230,0.5);}
.hover\\\\:border-pink:hover{--un-border-opacity:1;border-color:rgba(245,189,230,var(--un-border-opacity));}
.text-text{--un-text-opacity:1;color:rgba(202,211,245,var(--un-text-opacity));}
@media (prefers-color-scheme: dark){
.dark\\\\:bg-base{--un-bg-opacity:1;background-color:rgba(36,39,58,var(--un-bg-opacity));}
.dark\\\\:text-text{--un-text-opacity:1;color:rgba(202,211,245,var(--un-text-opacity));}
}
@media (prefers-color-scheme: light){
.light\\\\:bg-base{--un-bg-opacity:1;background-color:rgba(36,39,58,var(--un-bg-opacity));}
}"
`;

exports[`Catppuccin colours with theme extender using presetMini > and without default variant 1`] = `
"/* layer: default */
.disabled\\\\:bg-macchiato-red\\\\/25:disabled{background-color:rgba(237,135,150,0.25);}
.not-disabled\\\\:bg-macchiato-red:not(:disabled){--un-bg-opacity:1;background-color:rgba(237,135,150,var(--un-bg-opacity));}
.not-visited\\\\:bg-mocha-blue:not(:visited){--un-bg-opacity:1;background-color:rgba(137,180,250,var(--un-bg-opacity));}
.visited\\\\:bg-mocha-flamingo:visited{--un-bg-opacity:1;background-color:rgba(242,205,205,var(--un-bg-opacity));}
.border-frappe-pink\\\\/50{border-color:rgba(244,184,228,0.5);}
.hover\\\\:border-frappe-pink:hover{--un-border-opacity:1;border-color:rgba(244,184,228,var(--un-border-opacity));}
.text-latte-text{--un-text-opacity:1;color:rgba(76,79,105,var(--un-text-opacity));}
@media (prefers-color-scheme: dark){
.dark\\\\:bg-mocha-base{--un-bg-opacity:1;background-color:rgba(30,30,46,var(--un-bg-opacity));}
.dark\\\\:text-mocha-text{--un-text-opacity:1;color:rgba(205,214,244,var(--un-text-opacity));}
}
@media (prefers-color-scheme: light){
.light\\\\:bg-latte-base{--un-bg-opacity:1;background-color:rgba(239,241,245,var(--un-bg-opacity));}
}"
`;
